File: frontend/src/types/conversation.ts
/**
 * 对话相关的类型定义
 */

/** 消息角色 */
export type MessageRole = 'user' | 'assistant' | 'system'

/** 消息状态 */
export type MessageStatus = 'sending' | 'sent' | 'completed' | 'failed'

/** 单条对话消息 */
export interface Message {
  /** 消息 ID */
  id: string
  /** 会话 ID */
  conversation_id: string
  /** 消息角色 */
  role: MessageRole
  /** 消息内容 */
  content: string
  /** 关联的 SQL（如果是查询类消息） */
  sql?: string
  /** 消息状态 */
  status: MessageStatus
  /** 创建时间 */
  created_at: string
}

/** 对话会话 */
export interface Conversation {
  /** 会话 ID */
  id: string
  /** 会话标题 */
  title: string
  /** 关联的数据源 ID */
  datasource_id?: number
  /** 消息数量 */
  message_count: number
  /** 创建时间 */
  created_at: string
  /** 最后活跃时间 */
  updated_at: string
}

/** 创建会话请求 */
export interface CreateConversationParams {
  /** 会话标题 */
  title?: string
  /** 关联数据源 ID */
  datasource_id?: number
}

/** 发送消息请求 */
export interface SendMessageParams {
  /** 消息内容（自然语言问题） */
  content: string
  /** 会话 ID */
  conversation_id?: string
  /** 关联数据源 */
  datasource_id?: number
}

File: backend/app/models/conversation.py
"""会话工作区模型——数据与基础设施层"""
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.dialects.mysql import LONGTEXT
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin


class Conversation(Base, TimestampMixin):
    """会话工作区"""

    __tablename__ = "conversations"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False, default="新对话")
    workspace_type: Mapped[str] = mapped_column(String(50), nullable=False, default="query")  # query / attribution
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="active")  # active / archived
    metadata_json: Mapped[str | None] = mapped_column(Text, nullable=True)

    messages: Mapped[list["ConversationMessage"]] = relationship(
        back_populates="conversation", cascade="all, delete-orphan", order_by="ConversationMessage.created_at"
    )


class ConversationMessage(Base, TimestampMixin):
    """会话消息"""

    __tablename__ = "conversation_messages"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    conversation_id: Mapped[int] = mapped_column(
        ForeignKey("conversations.id", ondelete="CASCADE"), nullable=False, index=True
    )
    role: Mapped[str] = mapped_column(String(20), nullable=False)  # user / assistant / system
    content: Mapped[str] = mapped_column(LONGTEXT, nullable=False)
    message_type: Mapped[str] = mapped_column(String(50), nullable=False, default="text")  # text / sql / result / chart
    metadata_json: Mapped[str | None] = mapped_column(Text, nullable=True)

    conversation: Mapped["Conversation"] = relationship(back_populates="messages")